group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1573,6 +1589,33 @@ private void updateGroupsByTopics(
                 }
             });
         }
+        if (!oldSubscribedTopicRegex.isEmpty()) {
+            oldSubscribedTopicRegex.forEach(regex -> {
+                groupsByRegex.computeIfPresent(regex, (__, groupIds) -> {
+                    groupIds.remove(groupId);
+                    return groupIds.isEmpty() ? null : groupIds;
+                });
+                Pattern pattern = Pattern.compile(regex);
+                for (String topicName : metadataImage.topics().topicsByName().keySet()) {
+                    if (pattern.matcher(topicName).matches()) {
+                        unsubscribeGroupFromTopic(groupId, topicName);
+                    }
+                }
+            });
+        }
+        if (!newSubscribedTopicRegex.isEmpty()) {
+            newSubscribedTopicRegex.forEach(regex -> {
+                groupsByRegex
+                        .computeIfAbsent(regex, __ -> new TimelineHashSet<>(snapshotRegistry, 1))
+                        .add(groupId);
+                Pattern pattern = Pattern.compile(regex);
+                for (String topicName : metadataImage.topics().topicsByName().keySet()) {
+                    if (pattern.matcher(topicName).matches()) {
+                        subscribeGroupToTopic(groupId, topicName);
+                    }
+                }
```

Comment:
I am not sure about this for two reasons: (1) Computing the list of topics is quite expensive so doing it here may not be the best place; and (2) It would not catch the changes as it applies it only when the regex is stored. I think that we need to discuss the high level approach. Take a look at my previous comment.

Reply:
Just one question, why is computing the list of topics to subscribe to expensive here? My guess is that metadataImage is something that we need to load from disk?

Reply:
@Phuc-Hong-Tran It is definitely expensive wherever it will be :). The major downside of doing it here is that when you reload the group coordinator state from the partition, you may have obsolete records containing regex which are not compacted yet. For those, we would compute the list for nothing.

Reply:
@dajac thanks for the explanation